M2/Macaulay2/packages/SumsOfSquares.m2

```diff
@@ -473,13 +473,14 @@ createSOSModel(Matrix,Matrix) := o -> (F,v) -> (
     n2 := n*(n+1)//2;
 
     -- monomials in vvT
-    vvT := entries(v* transpose v);
-    mons := g -> set first entries monomials g;
-    K1 := toList \\ sum \\ mons \ flatten vvT;
+    vvTmatrix := v* transpose v;
+    vvT := entries vvTmatrix;
+    mons := flatten @@ entries @@ monomials @@ flatten;
```

Comment:
Also, I don't think you need to change so many things here, just change `set` to `unique`, which preserves order.

Reply:
Good point about the excessive use of `@@`'s.  :)

The `sum` on the next line was for taking a union of the sets and not addition, so it would take a little more than just swapping in `unique` here.  I think my current proposal is probably okay (without the `@@`'s) -- it calls `monomials` just once on the matrix instead of once per entry of the matrix and then gluing them together.

However, it still doesn't fix the build failure on the Ubuntu build...